File: easy/python/Question2441.py
from typing import List


class Solution:
    """
    方法一：枚举右维护左+哈希表
        时间复杂度O(n)
        空间复杂度O(n)
    """

    def findMaxK(self, nums: List[int]) -> int:
        idx = {}
        ans = -1
        for j, x in enumerate(nums):
            if -1 * x in idx:
                ans = max(ans, abs(x))
            idx[x] = j
        return ans


print(Solution().findMaxK([-10, 8, 6, 7, -2, -3]))

File: cpp/easy/Question2441.cpp
#include <cmath>
#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
   public:
    int findMaxK(vector<int>& nums) {
        unordered_set<int> s;
        int ans = -1;
        for (int i = 0; i < nums.size(); i++) {
            if (s.find(-1 * nums[i]) != s.end()) {
                ans = max(ans, abs(nums[i]));
            }
            s.insert(nums[i]);
        }
        return ans;
    }
};

int main() {
    vector<int> nums = {-1, 2, -3, 3};
    cout << Solution().findMaxK(nums) << endl;
}
